testsuite/python/dpd_press_contrib.py

```diff
@@ -0,0 +1,103 @@
+#
+# Copyright (C) 2013-2025 The ESPResSo project
+#
+# This file is part of ESPResSo.
+#
+# ESPResSo is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ESPResSo is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+import numpy as np
+import unittest as ut
+import unittest_decorators as utx
+import espressomd
+
+
+@ut.skipIf(espressomd.has_features(["DPD"]), "check default dictionaries")
+class NoDPDPressContrib(ut.TestCase):
+    system = espressomd.System(box_l=3 * [10])
+    system.time_step = 0.01
+    system.cell_system.skin = 0.4
+
+    n_part = 100
+
+    np.random.seed(seed=42)
+    system.part.add(pos=system.box_l * np.random.random((n_part, 3)))
+
+    def test_nondpd_case(self):
+        self.assertFalse(espressomd.has_features(["DPD"]))
+        energy_dict = self.system.analysis.energy()
+        self.assertNotIn("dpd", energy_dict,
+                         "'dpd' key should not be in the energy dictionary")
+
+        pressure_dict = self.system.analysis.pressure()
+        self.assertNotIn("dpd", pressure_dict,
+                         "'dpd' key should not be in the pressure dictionary")
+
+        pressure_tensor_dict = self.system.analysis.pressure_tensor()
+        self.assertNotIn("dpd", pressure_tensor_dict,
+                         "'dpd' key should not be in the pressure-tensor dictionary")
+
+
+@utx.skipIfMissingFeatures("DPD")
+class DPDPressContrib(ut.TestCase):
```

Comment:
Wouldn't it be easier to instead extend the existing DPD pressure test?

https://github.com/espressomd/espresso/blob/5c7ae80bbe4943c157e4e3940c61a60700f00295/testsuite/python/dpd.py#L395-L399

One line to get the DPD pressure via `system.analysis.pressure_tensor()["DPD"]` and one line to check it's equal to `-stress` (pressure should be the negative of the stress).